templates/CRM/Contact/Form/Edit/TagsAndGroups.tpl

```diff
@@ -12,24 +12,24 @@
   <div class="crm-accordion-header">{$title}</div>
   <div class="crm-accordion-body" id="tagGroup">
 {/if}
-    <table class="form-layout-compressed{if $context EQ 'profile'} crm-profile-tagsandgroups{/if}">
+    <table class="form-layout-compressed{if isset($context) && $context EQ 'profile'} crm-profile-tagsandgroups{/if}">
       <tr>
-        {if !$type || $type eq 'tag'}
+        {if empty($type) || (isset($type) && $type eq 'tag')}
           <td>
             <div class="crm-section tag-section">
-              {if $title}{$form.tag.label}<br>{/if}
+              {if !empty($title)}{$form.tag.label}<br>{/if}
               {$form.tag.html}
             </div>
-            {if $context NEQ 'profile'}
+            {if !isset($context) || (isset($context) && $context NEQ 'profile')}
```

Comment:
The second guard is redundant because after the `||` we know `$context` is set.
```suggestion
            {if !isset($context) || $context NEQ 'profile'}
```